### PR TITLE
[`tests`] Adds more tests + fix failing tests

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

This PRs adds more tests on the library. Here we extensively test enc-dec models in addition to the decoder models testing suite.
I also refactor tests to make them more clean

It seems that `generate` is not supported for seq2seq models except lora

cc @pacman100 

Also before this PR some tests realted with merged layers were actually failing, this PR fix them all

### CODE REVIEW

File: tests/test_encoder_decoder_models.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            {
+                "model_ids": PEFT_ENCODER_DECODER_MODELS_TO_TEST,
+                "lora_kwargs": {"init_lora_weights": [False], "merge_weights": [True, False]},
+            },
+        )
+    )
+    def test_merge_layers(self, test_name, model_id, config_cls, config_kwargs):
+        self._test_merge_layers(model_id, config_cls, config_kwargs)
+
+    # skip non lora models - generate does not work for prefix tuning, prompt tuning
```
Comment (pacman100) [inline tests/test_encoder_decoder_models.py:83]: `generate` should work for the prefix tuning. It won't work only for prompt tuning and p-tuning.

**Review by pacman100 — COMMENTED** [review]
This is great work and I like the way tests are refactored, very neat! Thanks a lot @younesbelkada 🚀

Left a comment.

File: tests/test_encoder_decoder_models.py (hunk 1 above)
Comment (younesbelkada) [reply tests/test_encoder_decoder_models.py:83]: Awesome let me dig into that!

Comment (younesbelkada) [reply tests/test_encoder_decoder_models.py:83]: Appears the way I was constructing the tests was wrong, pushed a fix

**Review by pacman100 — APPROVED** [review]
Thank you @younesbelkada for iterating, LGTM! 🚀🤗✨

File: tests/test_encoder_decoder_models.py (hunk 1 above)
Comment (pacman100) [reply tests/test_encoder_decoder_models.py:85]: nit: update the comment to exclude prefix tuning?

**Pull request merged**